react/getting_started_with_react/rendering_techniques.md

```diff
@@ -0,0 +1,288 @@
+### Introduction
+
+Now that we have learned how JSX works and how to write it. This lesson will cover how we can render multiple
+elements and conditionally render UI in JSX.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*  Rendering a list of elements in JSX
+*  Rendering components instead of elements
+*  How you can conditionally render UI
+
+### Rendering a List of Elements in JSX
+
+Let us say we want to create a component that lists multiple animals. We can do so by doing:
+
+~~~javascript
+function App() {
+    return (
+        <div>
+            <h1>Animals: </h1>
+            <ul>
+                <li>Lion</li>
+                <li>Cow</li>
+                <li>Snake</li>
+                <li>Lizard</li>
+            </ul>
+        </div>
+    )
+}
+~~~
+
+It is perfectly acceptable, but what if we want to render more than just four? It can be tedious and long, and most of the time, we will be dealing with a list rather than hard coding each animal. We can simplify this process by using a list of which we can apply the `map` method to map over the list. You have previously learned that we can embed expressions inside JSX with curly braces. So let us do just that:
+
+~~~javascript
+function App() {
+    const animals = ["Lion", "Cow", "Snake", "Lizard"];
+    return (
+        <div>
+            <h1>Animals: </h1>
+            <ul>
+                {animals.map((animal) => {
+                    return <li key={animal}>{animal}</li>
+                })}
+            </ul>
+        </div>
+    )
+}
+~~~
+
+We have defined an array called `animals` which contains what we have previously written. Now inside our JSX, we used `map` to map over each animal in the list and return the `li` element, adding `animal` as its text. It should now render the same as the previous snippet we wrote. It is now more dynamic and easy to handle.
+
+You may be curious as to what the `key` is in our `<li>` element. We will dive into how keys work in the next lesson. But, to explain briefly, It is to let React know the identity of each element in the list, React must know this information if you are dealing with a dynamic list where you add or remove elements. Since we are only dealing with a static list, it does not matter for now.
+
+### Rendering a Different Component
```

Comment:
That section also has some missing content, because in the `map`, I'm still using the `li` element, and should be extracted in a different component called `ListItem`, making it more consistent with the section title. Thanks!